Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -23,50 +23,54 @@ double BatteryStateOfChargeService::totalAmpHoursUsed() const
 }
 
 bool BatteryStateOfChargeService::isCharging() const
-// returns true if current is positive
+// returns true if current is negative
 {
-    if (current_ < 0)
+    if (currentCurrent_ < 0)
     {
-        return false;
+        return true;
     }
 
-    return true;
+    return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 // calculates time (in hours) and returns it in QTime
 {
-    double time;
+    double hours;
     if (isCharging())
     {
-        time = ampHoursUsed_ / current_;
+        hours = ampHoursUsed_ / (currentCurrent_ * -1);
```

Comment:
There is a nice function that will take care of the negative one for you in QT documentation.